src/test/java/emissary/server/api/EmissaryApiTest.java

```diff
@@ -289,4 +294,50 @@ void apiutils() throws Exception {
             Namespace.unbind("DirectoryPlace");
         }
     }
+
+    @Test
+    void configs() {
+        try (Response response = target("configuration/node").request().get()) {
+            assertEquals(200, response.getStatus());
+            ConfigsResponseEntity entity = response.readEntity(ConfigsResponseEntity.class);
+            assertTrue(entity.getErrors().isEmpty());
+            assertFalse(entity.getLocal().getConfigs().isEmpty());
+            assertEquals(1, entity.getLocal().getConfigs().size());
+            assertEquals("[CLUSTER]", entity.getLocal().getConfigs().get(0).getFlavors().toString());
+            assertEquals("[node.cfg, node-CLUSTER.cfg]", entity.getLocal().getConfigs().get(0).getConfigs().toString());
+        }
+    }
+
+    @Test
+    void configsDetailed() {
+        try (Response response = target("configuration/detailed/node").request().get()) {
+            assertEquals(200, response.getStatus());
+            ConfigsResponseEntity entity = response.readEntity(ConfigsResponseEntity.class);
+            assertTrue(entity.getErrors().isEmpty());
+            assertFalse(entity.getLocal().getConfigs().isEmpty());
+            assertEquals(3, entity.getLocal().getConfigs().size());
+            // config 1
+            assertEquals("[]", entity.getLocal().getConfigs().get(0).getFlavors().toString());
+            assertEquals("[node.cfg]", entity.getLocal().getConfigs().get(0).getConfigs().toString());
+            // config 2
+            assertEquals("[CLUSTER]", entity.getLocal().getConfigs().get(1).getFlavors().toString());
+            assertEquals("[node-CLUSTER.cfg]", entity.getLocal().getConfigs().get(1).getConfigs().toString());
+            // combined configs
+            assertEquals("[CLUSTER]", entity.getLocal().getConfigs().get(2).getFlavors().toString());
+            assertEquals("[node.cfg, node-CLUSTER.cfg]", entity.getLocal().getConfigs().get(2).getConfigs().toString());
+        }
+    }
+
+    private static final class TestEmissaryNode extends EmissaryNode {
+        public TestEmissaryNode() {
+            super();
+            this.nodeName = "TestNode";
+            this.nodePort = TEST_PORT;
```

Comment:
Is it necessary to set `this.nodeName` and `this.nodePort` when the result of the methods are mocked?